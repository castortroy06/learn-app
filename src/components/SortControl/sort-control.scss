@import '../../sass/bootstrap/variables';

.sort-control {
    background-color: $gray-300;
    letter-spacing: 0.889px;
    text-transform: uppercase;
    display: flex;

    &--label {
        color: $white;
        letter-spacing: 0.889px;
        opacity: 0.6027;
        margin-right: 1.9375rem;
    }

    &--dropdown {
        color: $white;
        border: none;
        background-color: transparent;
        position: relative;
        min-width: 160px;

        &::after {
            content: url('./img/arrow.svg');
            position: absolute;
            top: 0;
            right: 0;
        }

        a {
            text-decoration: none;
            font-weight: 500;
            border: none;
            color: $white;
            display: none;
            padding-right: 1.5rem;
            position: relative;
            z-index: 1;
            text-align: right;

            &.active {
                display: block;
            }
        }

        &.expanded {
            a {
                display: block;
            }
        }
    }
}
