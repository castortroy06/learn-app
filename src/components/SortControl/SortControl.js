import React, { useState } from 'react';
import './sort-control.scss';

const SortControl = ({ onChange }) => {
    const options = ['Release Date', 'Title'];
    const [active, setActive] = useState(options[0]);

    const handleClick = (e) => {
        e.preventDefault();
        const element = e.target;
        element.parentElement.classList.toggle('expanded');
        setActive(element.textContent);
        onChange(element.textContent);
    };

    return (
        <div className="sort-control">
            <p className="sort-control--label">Sort by</p>
            <div data-testid="sort-control-dropdown" className="sort-control--dropdown" id="sort-control" name="sort-control">
                {options.map((option, key) => (
                    <a data-testid={`sort-control-${option}`} key={key} href="/" onClick={(e) => handleClick(e)} className={option === active ? 'active' : ''}>{option}</a>
                ))}
            </div>
        </div>
    );
};

export default SortControl;
