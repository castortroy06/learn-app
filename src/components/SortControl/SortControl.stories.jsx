import SortControl from './SortControl';

export default {
    title: 'Sort Control',
}

const handleSubmit = (value) => {

};

export const Default = () => <SortControl onChange={handleSubmit} />
