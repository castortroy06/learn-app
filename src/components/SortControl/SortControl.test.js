import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

import SortControl from './SortControl';

describe('SortControl', () => {
    it('expect click on element expands dropdown', () => {
        render(<SortControl onChange={() => {}} />);

        const sortControlReleaseDateLink = screen.getByTestId('sort-control-Release Date');
        const sortControlReleaseTitle = screen.getByTestId('sort-control-Title');
        const sortControlDropdown = screen.getByTestId('sort-control-dropdown');
      
        // Click the button again and check if the "active" class is removed
        userEvent.click(sortControlReleaseDateLink);
        expect(sortControlDropdown).toHaveClass('expanded');
        expect(sortControlReleaseTitle).toBeVisible();

        userEvent.click(sortControlReleaseTitle);
        expect(sortControlDropdown).not.toHaveClass('expanded');
    });

    it('match snapshot', () => {
        const tree = renderer.create(<SortControl onChange={() => {}} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
