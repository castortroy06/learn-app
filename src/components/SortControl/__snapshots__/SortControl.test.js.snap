// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortControl match snapshot 1`] = `
<div
  className="sort-control"
>
  <p
    className="sort-control--label"
  >
    Sort by
  </p>
  <div
    className="sort-control--dropdown"
    data-testid="sort-control-dropdown"
    id="sort-control"
    name="sort-control"
  >
    <a
      className="active"
      data-testid="sort-control-Release Date"
      href="/"
      onClick={[Function]}
    >
      Release Date
    </a>
    <a
      className=""
      data-testid="sort-control-Title"
      href="/"
      onClick={[Function]}
    >
      Title
    </a>
  </div>
</div>
`;

exports[`SortControl renders all elements successfully 1`] = `
<div
  className="sort-control"
>
  <p
    className="sort-control--label"
  >
    Sort by
  </p>
  <div
    className="sort-control--dropdown"
    data-testid="sort-control-dropdown"
    id="sort-control"
    name="sort-control"
  >
    <a
      className="active"
      data-testid="sort-control-Release Date"
      href="/"
      onClick={[Function]}
    >
      Release Date
    </a>
    <a
      className=""
      data-testid="sort-control-Title"
      href="/"
      onClick={[Function]}
    >
      Title
    </a>
  </div>
</div>
`;
